Author: Martin Pitt <mpitt@debian.org>
Description: Disable PostgreSQL's automagic path mangling and fix libdir for pg_config, so that pg_config in /usr/bin and /usr/lib/postgresql/8.3/bin behave identically.
Forwarded: No, Debian specific
Bug-Debian: http://bugs.debian.org/462037

--- postgresql-9.0-9.0~alpha5/src/bin/pg_config/pg_config.c	2010-01-02 17:57:58.000000000 +0100
+++ postgresql-9.0-9.0~alpha5.new/src/bin/pg_config/pg_config.c	2010-04-30 12:57:42.726134778 +0200
@@ -26,6 +26,8 @@
 
 #include "port.h"
 
+#include "../port/pg_config_paths.h"
+
 static const char *progname;
 static char mypath[MAXPGPATH];
 
@@ -463,7 +465,6 @@
 {
 	int			i;
 	int			j;
-	int			ret;
 
 	set_pglocale_pgservice(argv[0], PG_TEXTDOMAIN("pg_config"));
 
@@ -479,13 +480,7 @@
 		}
 	}
 
-	ret = find_my_exec(argv[0], mypath);
-
-	if (ret)
-	{
-		fprintf(stderr, _("%s: could not find own program executable\n"), progname);
-		exit(1);
-	}
+	snprintf(mypath, sizeof(mypath), "%s/%s", PGBINDIR, progname);
 
 	/* no arguments -> print everything */
 	if (argc < 2)
